# Twitter Sentiment Analysis

In [92]:
# Regex library
import re

# Read from the positive word file and make it into a list.
positive = open('positive.txt','r')
positive_data = positive.readlines()

# Remove new line(/n) from the data list.
positive_list = [i.strip() for i in positive_data]
print(positive_list)

['a+', 'abound', 'abounds', 'abundance', 'abundant', 'accessable', 'accessible', 'acclaim', 'acclaimed', 'acclamation', 'accolade', 'accolades', 'accommodative', 'accomodative', 'accomplish', 'accomplished', 'accomplishment', 'accomplishments', 'accurate', 'accurately', 'achievable', 'achievement', 'achievements', 'achievible', 'acumen', 'adaptable', 'adaptive', 'adequate', 'adjustable', 'admirable', 'admirably', 'admiration', 'admire', 'admirer', 'admiring', 'admiringly', 'adorable', 'adore', 'adored', 'adorer', 'adoring', 'adoringly', 'adroit', 'adroitly', 'adulate', 'adulation', 'adulatory', 'advanced', 'advantage', 'advantageous', 'advantageously', 'advantages', 'adventuresome', 'adventurous', 'advocate', 'advocated', 'advocates', 'affability', 'affable', 'affably', 'affectation', 'affection', 'affectionate', 'affinity', 'affirm', 'affirmation', 'affirmative', 'affluence', 'affluent', 'afford', 'affordable', 'affordably', 'afordable', 'agile', 'agilely', 'agility', 'agreeable', 'ag

In [93]:
# Similarly fetch the negative words from the file and make it into a list.
negative = open('negative.txt','r')
negative_data = negative.readlines()

# Remove new line (/n) from the list
negative_list = [i.strip() for i in negative_data]
print(negative_list)

['2-faced', '2-faces', 'abnormal', 'abolish', 'abominable', 'abominably', 'abominate', 'abomination', 'abort', 'aborted', 'aborts', 'abrade', 'abrasive', 'abrupt', 'abruptly', 'abscond', 'absence', 'absent-minded', 'absentee', 'absurd', 'absurdity', 'absurdly', 'absurdness', 'abuse', 'abused', 'abuses', 'abusive', 'abysmal', 'abysmally', 'abyss', 'accidental', 'accost', 'accursed', 'accusation', 'accusations', 'accuse', 'accuses', 'accusing', 'accusingly', 'acerbate', 'acerbic', 'acerbically', 'ache', 'ached', 'aches', 'achey', 'aching', 'acrid', 'acridly', 'acridness', 'acrimonious', 'acrimoniously', 'acrimony', 'adamant', 'adamantly', 'addict', 'addicted', 'addicting', 'addicts', 'admonish', 'admonisher', 'admonishingly', 'admonishment', 'admonition', 'adulterate', 'adulterated', 'adulteration', 'adulterier', 'adversarial', 'adversary', 'adverse', 'adversity', 'afflict', 'affliction', 'afflictive', 'affront', 'afraid', 'aggravate', 'aggravating', 'aggravation', 'aggression', 'aggress

In [94]:
# Get the stop words from the file and make it into a list.
stopwords= open('stopwords.txt','r')
stopwords_data = stopwords.readlines()

# Remove new line(/n) from the list.
stopwords_list = [i.strip() for i in stopwords_data]
print(stopwords_list)

['a', 'about', 'above', 'across', 'after', 'afterwards', 'again', 'against', 'all', 'almost', 'alone', 'along', 'already', 'also', 'although', 'always', 'am', 'among', 'amongst', 'amoungst', 'amount', 'an', 'and', 'another', 'any', 'anyhow', 'anyone', 'anything', 'anyway', 'anywhere', 'are', 'around', 'as', 'at', 'b', 'back', 'be', 'became', 'because', 'become', 'becomes', 'becoming', 'been', 'before', 'beforehand', 'behind', 'being', 'below', 'beside', 'besides', 'between', 'beyond', 'bill', 'both', 'bottom', 'but', 'by', 'c', 'call', 'can', 'cannot', 'cant', 'co', 'computer', 'con', 'could', 'couldnt', 'd', 'de', 'describe', 'detail', 'do', 'done', 'down', 'due', 'during', 'e', 'each', 'eg', 'eight', 'either', 'eleven', 'else', 'elsewhere', 'etc', 'even', 'ever', 'every', 'everyone', 'everything', 'everywhere', 'except', 'f', 'few', 'fifteen', 'fify', 'fill', 'find', 'fire', 'first', 'five', 'for', 'former', 'formerly', 'forty', 'found', 'four', 'from', 'front', 'full', 'further', 'g

In [95]:
# Check if there is any common terms in the negative or positive list. Expecting no values to be returned
intersection_postive_negative_word = set(positive_list) & set(negative_list)
print(intersection_postive_negative_word)

set()


In [96]:
# Check if there is any common terms in the positive or stop_word list. Expecting no values to be returned
intersection_postive_stop_word = set(positive_list) & set(stopwords_list)
print(intersection_postive_stop_word)

set()


In [97]:
# Check if there is any common terms in the negative or stop_word list. Expecting no values to be returned
intersection_negative_stop_word = set(negative_list) & set(stopwords_list)
print(intersection_negative_stop_word)

set()


## Reading tweets with #Trump

In [98]:
# Read the file in UTF-8 format.
#Cleaning the dataset
with open("Trump.txt",'r',encoding='utf-8') as file:
    tweets = file.readlines()
    for item in tweets:
     # Removing the URLs, unwanted symbols and user names using regex and convert it to list format
     tweet_lowcase = item.lower()
     cleaning_data = ' '.join(re.sub("(@[A-Za-z0-9]+)|([^0-9A-Za-z \t])|(\w+:\/\/\S+)", " ", tweet_lowcase).split())
     # Remove the emojis by comparing the character and checking if it is a valid digit.
     remove_emoji = ' '.join(s for s in cleaning_data.split() if not any(c.isdigit() for c in s))
     # Convert the formatted string to list.
     tweet_list = list(remove_emoji.split(" "))
    print(tweet_list)

['b', 'rt', 'years', 'after', 'the', 'comms', 'dept', 'declared', 'war', 'on', 'today', 'p', 'n', 'b', 'rt', 'mlk', 'said', 'judge', 'frank', 'johnson', 'gave', 'meaning', 'to', 'word', 'his', 'rulings', 'protected', 'selma', 'marchers', 'amp', 'desegregated', 'm', 'n', 'b', 'rt', 'we', 'learned', 'anything', 'in', 'the', 'trump', 'era', 'it', 'is', 'that', 'our', 'rights', 'our', 'freedom', 'our', 'democracy', 'require', 'everyone', 'to', 'do', 'their', 'n', 'b', 'rt', 'hey', 'when', 'you', 'attend', 'a', 'racist', 'kkk', 'member', 's', 'funeral', 'and', 'treat', 'old', 'robert', 'byrd', 'like', 'a', 'statesman', 'maybe', 'you', 'shouldn', 'n', 'b', 'y', 'all', 'already', 'know', 'this', 'but', 'paula', 'white', 'is', 'a', 'toxic', 'fraud', 'and', 'a', 'political', 'idolator', 'blech', 'n', 'b', 'dershowitz', 'will', 'harm', 'any', 'exculpatory', 'case', 'for', 'trump', 'by', 'trying', 'to', 'quash', 'the', 'truth', 'facts', 'evidence', 'in', 'a', 'trial', 'this', 'the', 'nyt', 'edito

Count the number of words in the list. Once we do that, let us check the validity of positive and negative words by checking if both of them any common words. This is to ensure that we don't treat a single word in both positive and negative sentiment.

In [99]:
# Count the total number of words.
count_total_words = len(tweet_list)
print('Total word count for #trump tweets:',count_total_words)

Total word count for #trump tweets: 25797


Using the lists that we created, compare it with the words in the tweet list and measure the sentiment.

In [100]:
# Initialize the counts.
positive_count =0
negative_count =0
stopword_count =0
other_word_count =0

# For each word, increment the count based on the list.
for word in tweet_list:
    if word in positive_list:
        positive_count+=1
    elif word in negative_list:
        negative_count+=1
    elif word in stopwords_list:
        stopword_count+=1
    else:
        other_word_count+=1
print('The positive count of words in the tweets list:',positive_count)
print('The negative count of words in the tweets list:',negative_count)
print('The stopword count of words in the tweets list:',stopword_count)
print('Words not in positive/negative/stop words:',other_word_count)

The positive count of words in the tweets list: 1534
The negative count of words in the tweets list: 868
The stopword count of words in the tweets list: 10981
Words not in positive/negative/stop words: 12414


Based on the count, measure the ratio of positive and negative words using the total word count.

In [101]:
# Positive ratio and percentage.
ratio_of_positive = positive_count/count_total_words
print('Positive ratio', round(ratio_of_positive, 3))
Percentage_ratio_of_positive = ratio_of_positive * 100
print('Positive ratio percentage: ', round(Percentage_ratio_of_positive, 2),'%')

Positive ratio 0.059
Positive ratio percentage:  5.95 %


In [102]:
# Negative ratio and percentage.
ratio_of_negative = negative_count/count_total_words
print('Negative ratio', round(ratio_of_negative, 3))
Percentage_ratio_of_negative = ratio_of_negative * 100
print('Negative ratio percentage: ', round(Percentage_ratio_of_negative, 2),'%')

Negative ratio 0.034
Negative ratio percentage:  3.36 %


In [103]:
sentiment_score_trump = positive_count - negative_count
print(sentiment_score_trump)
if sentiment_score_trump > 0:
    print("Overall sentiment is positive")
elif sentiment_score_trump < 0:
    print("Overall sentiment is negative")
elif sentiment_score_trump == 0:
     print("Overall sentiment is neutral")

666
Overall sentiment is positive


## Conclusion:

Based on the above calculation, it is clear that the overall sentiment is positive but it is only mildly positive since the negative ratio percentage is 3.36% vs the positive ratio percentage which is 5.95%.

## Read tweets with Obama

In [104]:

with open("obama.txt",'r',encoding='utf-8') as file:
    tweet_obama = file.readlines()
    for item in tweet_obama:
      # Removing the URLs, unwanted symbols , usernames etc and convert it to list format by using regex.
      tweet_lowcase = item.lower()
      cleaning_data = ' '.join(re.sub("(@[A-Za-z0-9]+)|([^0-9A-Za-z \t])|(\w+:\/\/\S+)", " ", tweet_lowcase).split())
      # Remove the emojis by comparing the character.
      remove_emoji = ' '.join(s for s in cleaning_data.split() if not any(c.isdigit() for c in s))
      # Convert the string to a list.
      tweet_obama_list = list(remove_emoji.split(" "))
    print(tweet_obama_list)

['b', 'rt', 'nadler', 'asks', 'why', 'trump', 'cared', 'about', 'the', 'massive', 'energy', 'concern', 'burisma', 'when', 'of', 'thousands', 'ukraine', 'companies', 'are', 'al', 'n', 'b', 'rt', 'miss', 'm', 'q', 'a', 'group', 'of', 'top', 'lawmakers', 'accused', 'hu', 'n', 'b', 'rt', 'sekulow', 'but', 'obama', 'held', 'up', 'aid', 'to', 'egypt', 'once', 'n', 'nagain', 'this', 'is', 'not', 'meant', 'to', 'make', 'an', 'actual', 'argument', 'he', 'said', 'this', 'knowing', 'n', 'b', 'pro', 'pedophilia', 'argument', 'by', 'arguing', 'that', 'unemployment', 'is', 'lower', 'than', 'it', 'is', 'under', 'obama', 'take', 'your', 'meds', 'schizo', 'n', 'b', 'rt', 'i', 'waiting', 'for', 'the', 'war', 'hawk', 'obama', 'statement', 'on', 'bernie', 'probably', 'coming', 'soon', 'iowa', 'is', 'close', 'and', 'they', 'see', 'the', 'momentum', 'n', 'b', 'rt', 'nadler', 'asks', 'why', 'trump', 'cared', 'about', 'the', 'massive', 'energy', 'concern', 'burisma', 'when', 'of', 'thousands', 'ukraine', 'com

Count the number of words in the list. Once we do that, let us check the validity of positive and negative words by checking if both of them any common words. This is to ensure that we don't treat a single word in both positive and negative sentiment.

In [105]:
count_total_word = len(tweet_obama_list)
print('Total word count for #obama tweets:',count_total_word)

Total word count for #obama tweets: 12301


Using the lists that we created, compare it with the words in the tweet list and measure the sentiment.

In [106]:
# Initialize the counts.
positive_count =0
negative_count =0
stopword_count =0
other_word_count =0

# For each word, increment the count based on the list.
for word in tweet_obama_list:
    if word in positive_list:
        positive_count+=1
    elif word in negative_list:
        negative_count+=1
    elif word in stopwords_list:
        stopword_count+=1
    else:
        other_word_count+=1
print('The positive count of words in the tweets list:',positive_count)
print('The negative count of words in the tweets list:',negative_count)
print('The stopword count of words in the tweets list:',stopword_count)
print('Words not in positive/negative/stop words:',other_word_count)

The positive count of words in the tweets list: 304
The negative count of words in the tweets list: 435
The stopword count of words in the tweets list: 5033
Words not in positive/negative/stop words: 6529


Based on the count, measure the ratio of positive and negative words using the total word count.

In [107]:
positive_ratio = positive_count/count_total_word
print('Positive ratio:', round(positive_ratio,3))
Percentage_positive_words = positive_ratio * 100
print('Positive percentage:', round(Percentage_positive_words,2),'%')

Positive ratio: 0.025
Positive percentage: 2.47 %


In [108]:
negative_ratio = negative_count/count_total_word
print('Negative ratio:', round(negative_ratio, 3))
Percentage_negative_words = negative_ratio * 100
print('Negative percentage:', round(Percentage_negative_words,2),'%')

Negative ratio: 0.035
Negative percentage: 3.54 %


In [109]:
sentiment_score = positive_count - negative_count
print(sentiment_score)
if sentiment_score > 0:
    print("Overall sentiment is positive")
elif sentiment_score < 0:
    print("Overall sentiment is negative")
elif sentiment_score == 0:
     print("Overall sentiment is neutral")

-131
Overall sentiment is negative


# Conclusion:

Based on the above calculation, it is clear that the overall sentiment for #obama tweets is negative but it is only mildly negative since the negative ratio percentage is 3.54% vs the positive ratio percentage which is 2.47%.